### ISSUE TITLE
Incorect memory calculation for containes with memory limit set in higher level of cgroup hierarchy

### ISSUE THREAD

**Issue by YuriGrigorov** [VictoriaMetrics/VictoriaMetrics#699]
**Version**
```
20200815-130846-tags-v1.40.0-0-ged00eb3f3
```

**Intro**

Memory cgroup has hierarchy support

From https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt

```
6. Hierarchy support

The memory controller supports a deep hierarchy and hierarchical accounting.
The hierarchy is created by creating the appropriate cgroups in the
cgroup filesystem. Consider for example, the following cgroup filesystem
hierarchy

           root
         /  |   \
        /   |    \
       a   b     c
                  | \
                  |  \
                  d   e

In the diagram above, with hierarchical accounting enabled, all memory
usage of e, is accounted to its ancestors up until the root (i.e, c and root),
that has memory.use_hierarchy enabled. If one of the ancestors goes over its
limit, the reclaim algorithm reclaims from the tasks in the ancestor and the
children of the ancestor.
```

**Describe the bug**

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/efc730863bb85b47e71c9f5d34238d257a1469d5/lib/memory/memory.go#L35

```sysTotalMemory()``` returns all system memory or value from ```/sys/fs/cgroup/memory/memory.limit_in_bytes``` if available.

If memory limit is set in higher level of cgroup hierarchy then value of lower cgoups is default ```9223372036854771712``` (as value in ```/sys/fs/cgroup/memory/memory.limit_in_bytes``` from inside of container). 
```
/sys/fs/cgroup/memory/c/vmagent/memory.limit_in_bytes: 2147483648 <== QUOTA
/sys/fs/cgroup/memory/c/vmagent/p/memory.limit_in_bytes: 9223372036854771712
/sys/fs/cgroup/memory/c/vmagent/p/libpod-bf7d/memory.limit_in_bytes: 9223372036854771712 <== MOUNTED TO CONTAINER
```
In this case ```allowedMemory``` and ```remainingMemory``` are calculated incorectly.

It is possible specify memory.allowedBytes. But if total available memory is calculated form default value, then remainingMemory is calculated incorectly.

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/efc730863bb85b47e71c9f5d34238d257a1469d5/lib/memory/memory.go#L42

```hierarchical_memory_limit``` is available in ```/sys/fs/cgroup/memory/memory.stat```
```
grep hierarchical_memory_limit /sys/fs/cgroup/memory/memory.stat
hierarchical_memory_limit 2147483648
```

But it will be nice to have some option to specify memory limit for cases when calculation is not possible to override value from ```/sys/fs/cgroup/memory/memory.limit_in_bytes```

**Comment by valyala** [comment]
@YuriGrigorov , could you build VictoriaMetrics components from be6ae4b5e755df57db135b77e749067c8822a96a (single-node version) or from d2f6f96e4a12ef6d3a1bbb51cb5579189f772f74 (cluster version) and verify whether they respect hierachical memory limits set in cgroups?

**Comment by valyala** [comment]
FYI, commits mentioned above have been included in [v1.40.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by YuriGrigorov** [comment]
> @YuriGrigorov , could you build VictoriaMetrics components from [be6ae4b](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/be6ae4b5e755df57db135b77e749067c8822a96a) (single-node version) or from [d2f6f96](https://github.com/VictoriaMetrics/VictoriaMetrics/commit/d2f6f96e4a12ef6d3a1bbb51cb5579189f772f74) (cluster version) and verify whether they respect hierachical memory limits set in cgroups?

```
# cat /sys/fs/cgroup/memory/memory.limit_in_bytes
9223372036854771712
# grep hierarchical_memory_limit /sys/fs/cgroup/memory/memory.stat
hierarchical_memory_limit 8589934592
```

Looks fine
```
VictoriaMetrics/lib/memory/memory.go:43 limiting caches to 5153960755 bytes, leaving 3435973837 bytes to the OS according to -memory.allowedPercent=60.000000
```

Thank you!
